src/transport_ofi.h

```diff
@@ -308,6 +310,17 @@ extern struct fid_ep* shmem_transport_ofi_target_ep;
             shmem_free_list_unlock(ctx->bounce_buffers);                        \
     } while (0)
 
+static inline
+void shmem_transport_probe(void) {
+#if defined(ENABLE_MANUAL_PROGRESS) && !defined(HARD_POLLING)
```

Comment:
I think `HARD_POLLING` here might be a looming bug.  This gets switched on automatically when XPMEM is enabled, so enabling XPMEM could have the second order effect of disabling manual progress.

Reply:
IIRC, the `target_cntr` is disabled when hard polling is enabled.  We might need to update that to hard polling enabled and manual progress disabled.

Reply:
Not sure I fully understand.  Do you mean that we should consider changing instances of
`#ifndef HARD_POLLING`
to
`#if !defined(HARD_POLLING) || defined(ENABLE_MANUAL_PROGRESS)`
which guard `target_cntr`?  If so, I'm not sure I completely appreciate the consequences of enabling hard polling and manual progress.

Reply:
The hard polling code was fairly well isolated before threading support was added, but it's less so now.  The `HARD_POLLING` parameter was originally meant to influence how shmem_wait operations are implemented -- do we poll memory or block on a transport-layer event (see shmem_synchronization.c:120).  Presently, when you enable hard polling, we don't need the `target_cntr`, so it gets compiled out (which makes the GNI provider happy, so it's important we continue to support this mode).  However, this PR adds a second case where we need the `target_cntr` -- manual progress.

Anyway, long winded way to say "yes" I think we need to update the `target_cntr` preprocessor magic so we get the cntr when needed to support either non-hard polling (which we still support in single-threaded runs without XPMEM) or manual progress.

Reply:
Ok, I just pushed commit 4d1b444 to try to address this.  I'm still not quite sure if things get screwy when  simultaneously enabling both hard polling and manual progress.  In that case, does `query_for_fabric()` look ok with the `RMA_EVENT` and `MR_RMA_EVENT` settings?  Thanks!

That commit also allows the probe even with `FI_THREAD_SAFE` - I figure it could be useful to try out later, but yeah, it's probably less important than with `FI_THREAD_COMPLETION`.